tasmota/xdrv_41_tcp_bridge.ino

```diff
@@ -118,6 +174,142 @@ void TCPLoop(void)
+	    }
+            */
+
+            AddLog(LOG_LEVEL_DEBUG, PSTR(D_LOG_TCP "Received unhandled telnet command: %02x"), c);
+
+            iac_level[i] = 0;
+            continue;
+
+          case 2:
+            if(willwontdodont[i] == TELNET_WILL and c == RFC2217_COMPORTOPTION) {
+              uint8_t comportoption_response[]={ TELNET_IAC, TELNET_DO, RFC2217_COMPORTOPTION };
+
+              AddLog(LOG_LEVEL_DEBUG, PSTR(D_LOG_TCP "RFC2217 response: %*_H"), sizeof(comportoption_response), comportoption_response);
+
+              client.write(comportoption_response, sizeof(comportoption_response));
+
+              iac_level[i] = 0;
+              continue;
+            }
+
+            AddLog(LOG_LEVEL_DEBUG, PSTR(D_LOG_TCP "Received unhandled telnet option: %02x %02x"), willwontdodont[i], c);
+
+            // other options currently ignored
+            iac_level[i] = 0;
+            continue;
+
+          case 3: // special sequence element 0
+          case 4:
+          case 5:
+          case 6:
+          case 7:
+          case 8:
+          case 9:
+            if( c == TELNET_SE ) { // here, special end; this is not properly scanning for the official IAC SE though, yet works
+
+              if( specialseq[i][0] == RFC2217_COMPORTOPTION ) {
+
+                uint32_t new_baudrate, new_serialconfig;
+
+                new_baudrate = GetSerialBaudrate();
+                new_serialconfig = serial_config_rfc2217;
+
+                if (specialseq[i][1] == 1) { // SET-BAUDRATE
+                  uint32_t baudrate = (((specialseq[i][2]*256)+specialseq[i][3])*256+specialseq[i][4])*256+specialseq[i][5];
+                  if (baudrate == 300 || (baudrate >= 1200 and baudrate <= 115200) ) { new_baudrate = baudrate; }
+                }
+                if (specialseq[i][1] == 2) { // SET-DATASIZE
+                  if (specialseq[i][2] >= 5 and specialseq[i][2] <=8 ) { 
+                    new_serialconfig = (new_serialconfig & 0xfc) | (specialseq[i][2] - 5);
+                  }
+                }
+                if (specialseq[i][1] == 3) { // SET-PARITY
+                  if (specialseq[i][2] >= 1 and specialseq[i][2] <=3 ) { 
+                    uint8_t map[4] = {0,0,2,1}; // map from -NOE to NEO-
+                    new_serialconfig = (new_serialconfig & 0xe7) | (map[specialseq[i][2]]<<3);
+                  }
+                }
+                if (specialseq[i][1] == 4) { // SET-STOPSIZE
+                  if (specialseq[i][2] >= 1 and specialseq[i][2] <=2 ) { // skip 3=1.5
+                    new_serialconfig = (new_serialconfig & 0xfb) | ((specialseq[i][2] - 1)<<2);
+                  }
+                }
+
+                if( new_baudrate != GetSerialBaudrate() || new_serialconfig != serial_config_rfc2217 ) { // only if changed
+                  serial_config_rfc2217 = new_serialconfig;
+                  TCPSerial->begin(new_baudrate, ConvertSerialConfig(new_serialconfig));  // Reinitialize serial port with new config
```

Comment:
So the new baudrate and config received from RFC2217 will not be saved as Settings ?
Just for the current session ?

Reply:
Exactly. Imho with RFC2217, that is for the use cases I know, the required parameters are set/changed on-the-fly.